### DISCUSSION TITLE
Compass widget

### DISCUSSION THREAD

**Discussion by nicostudt** [ROS-Mobile/ROS-Mobile-Android#77] [Ideas]
See initial post at #75.

**Comment by nicostudt** [comment]
Okay, then lets discuss a little.
At first: nice work you have done so far! Your own widgets look very promising.

I would suggest to create a Compass Widget or whatever name would fit more to display angle values, so you dont have to send the compass as an image. At the moment, ROS is not very efficient in terms of network stability when transmitting image streams. 

http://docs.ros.org/en/api/geometry_msgs/html/msg/Twist.html I think is the right way to go. So we could potentially implement a widget subscribing to x Twist publishers and displaying them as a "compass".

**Reply by MoffKalast** [reply]
Yeah efficiency wasn't exactly my goal here, just something quick and dirty so I could have those indicators there quickly to hit a testing deadline. I'm sure there could be a much better actual native widget for it, but it would take longer to make. Besides, I'm not really a fan of coding in android studio and native android in general myself. And I'm sure there are others who may take the same approach.

ROS should be pretty ok at sending images assuming you use the less lavish [CompressedImage](https://docs.ros.org/en/api/sensor_msgs/html/msg/CompressedImage.html) which would effectively switch a raw stream to an mjpeg stream (though there's no support for it in the app yet from what I can tell), and I also used something like 100x100 so it wasn't much data in total. Or webrtc_ros for h264 mp4 video but that's way overkill for this sort of thing. The only place where it's best use raw video is in editing recorded footage 😛
